File: BackEnd/Application/UseCases/Users/CreateUser/CreateUserCommandHandler.cs
using Domain.Aggregates;
using FluentValidation;
using Infrastructure.Repository;
using Infrastructure.UnitOfWork;
using MediatR;

namespace Application.UseCases.Users.CreateUser;

public class CreateUserCommandHandler(
    IRepository<User, Guid> userRepository,
    IUnitOfWork unitOfWork,
    IValidator<CreateUserCommand> validator
) : IRequestHandler<CreateUserCommand, Guid>
{
    public async Task<Guid> Handle(CreateUserCommand request, CancellationToken cancellationToken)
    {
        var validateResult = await validator.ValidateAsync(request, cancellationToken);
        if (!validateResult.IsValid)
        {
            throw new ValidationException(validateResult.Errors);
        }
        
        var user = User.Create(request.FullName, request.Email);
        await userRepository.AddAsync(user, cancellationToken);
        await unitOfWork.SaveChangesAsync(cancellationToken);
        return user.Id;
    }
}

File: flutter/lib/application/use_cases/users/create_user/create_user_command_handler.dart
import 'package:enterprise_management/application/use_cases/users/create_user/create_user_command.dart';
import 'package:enterprise_management/infrastructure/repositories/user_repository.dart';
import 'package:enterprise_management/shared_kernel/cqrs/command_handler.dart';
import 'package:enterprise_management/shared_kernel/result/result.dart';

class CreateUserCommandHandler extends ICommandHandler<CreateUserCommand, String> {
  const CreateUserCommandHandler({required this._userRepository});
  final IUserRepository _userRepository;

  @override
  Future<Result<String>> handle(CreateUserCommand command) async {
    final res = await _userRepository.createUser(fullName: command.fullName, email: command.email);
    return Result.success(res);
  }

}